### PR TITLE
[DOCS] Docs for creating tests with `nf-test`

### LINKED ISSUE #116: Update documentation for tests using nf-test
This task is done in 2 parts :

- [x] Remove old test documentation ~~, replace by a note containing a link to the documentation in the alpha release on github~~
- [ ] Create step-by-step documentation for test creation with `nf-test `

### PR DESCRIPTION
**Pull request by gagnonanthony**
## Type of improvement

**If submitting a new module or fixing a bug, please use the appropriate template.**

- [x] Documentation
- [ ] Development tools (e.g. linter, formatter, etc.)
- [ ] Development container
- [ ] Global update (please specify)
- [ ] Other (please specify)

## Describe your improvement

I have added a description and walkthrough on how to write a test case with nf-test. It was long overdue (my bad!, closes #116 ). 

## Checklist before requesting a review

- [ ] Ensure the syntax is correct (**EditorConfig** and **Prettier** must pass)
- [ ] Run the test suites if your changes affect any module
- [ ] Regenerate the **Poetry** lock file if you have updated the dependencies
- [x] Ensure the documentation is up-to-date

### CODE REVIEW

**Review by ThoumyreStanislas — APPROVED** [review]
Better than Shakespeare, the story is deep and the twists unexpected, bravo! I'm kidding but nothing to say for myself, i find it really understandable! good work!

File: docs/MODULE.md [hunk 1]
```diff
@@ -151,6 +152,23 @@ already follow all guidelines. You will find related files in :
 
   - Call `touch <file>` to generate empty files for all required outputs.
 
+### Editing `./modules/nf-scil/<category>/>tool>/environment.yml`:
```
Comment (AlexVCaron) [inline docs/MODULE.md:155]:
```suggestion
### Editing `./modules/nf-scil/<category>/<tool>/environment.yml`:
```

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
:fire: great job ! A little typo, then good to go :ship:

**Review by AlexVCaron — APPROVED** [review]

**Comment by AlexVCaron** [comment]
If there is nothing more to say, then LGTM ! :rocket:

**Pull request merged**